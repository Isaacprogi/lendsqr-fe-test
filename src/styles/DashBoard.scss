@import '../styles/variables/mixins';



.dash-board{
    height: 100%;
    padding-right: 1rem;
    padding-left: 1rem;
    overflow-x: hidden;

    @include smallDevices{
        padding-left: 2rem;
        padding-right: 2rem;
    }

  
    h1{
     font-family: 'Work Sans';
     font-style: normal;
     font-weight: 500;
     font-size: 24px;
     color: #213F7D;          
     margin-top: 3rem;
     margin-bottom: 3rem;
    }

           

     @include smallDevices {
         flex-direction: row;
         justify-content: space-between;
         align-items: flex-start;
      }

      

             
}  

