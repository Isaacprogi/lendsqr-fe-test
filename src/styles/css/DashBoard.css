.dash-board {
  height: 100%;
  padding-right: 1rem;
  padding-left: 1rem;
  overflow-x: hidden;
}
@media screen and (min-width: 568px) {
  .dash-board {
    padding-left: 2rem;
    padding-right: 2rem;
  }
}
.dash-board h1 {
  font-family: "Work Sans";
  font-style: normal;
  font-weight: 500;
  font-size: 24px;
  color: #213F7D;
  margin-top: 3rem;
  margin-bottom: 3rem;
}
@media screen and (min-width: 568px) {
  .dash-board {
    -webkit-box-orient: horizontal;
    -webkit-box-direction: normal;
        -ms-flex-direction: row;
            flex-direction: row;
    -webkit-box-pack: justify;
        -ms-flex-pack: justify;
            justify-content: space-between;
    -webkit-box-align: start;
        -ms-flex-align: start;
            align-items: flex-start;
  }
}/*# sourceMappingURL=DashBoard.css.map */