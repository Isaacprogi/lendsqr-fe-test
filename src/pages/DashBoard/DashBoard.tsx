import '../../styles/css/DashBoard.css'

export const DashBoard = () => {


    return (
        <div className="dash-board">
                        
                        <h1>DashBoard</h1>
       </div>
    )
}
