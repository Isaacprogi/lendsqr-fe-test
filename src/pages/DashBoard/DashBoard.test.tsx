import { render } from '@testing-library/react';
import { describe, it, test, vi } from 'vitest';
import { DashBoard } from './DashBoard';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

describe('DashBoard', () => {
    const dropDownActive = false;
    const setShowSideBar = vi.fn();
    const showSideBar = false;

    it('renders DashBoard component', () => {
        const { getByText } = render(
            <DashBoard
                dropDownActive={dropDownActive}
                setShowSideBar={setShowSideBar}
                showSideBar={showSideBar}
            />
        );

        expect(getByText('DashBoard')).toBeInTheDocument();
    });

    test('should match snapshot', async () => {
        const tree = renderer.create(
            <BrowserRouter>
                <DashBoard
                    dropDownActive={dropDownActive}
                    showSideBar={showSideBar}
                    setShowSideBar={setShowSideBar}
                />
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
