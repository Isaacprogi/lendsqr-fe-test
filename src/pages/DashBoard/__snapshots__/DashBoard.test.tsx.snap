// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DashBoard > should match snapshot 1`] = `
<div
  className="dash-board"
>
  <h1>
    DashBoard
  </h1>
</div>
`;
